setup.py

```diff
@@ -169,34 +169,23 @@ def function_three():
         'License :: OSI Approved :: MIT License',
         'Operating System :: OS Independent',
         'Programming Language :: Python',
-        'Programming Language :: Python :: 2',
-        'Programming Language :: Python :: 2.7',
-        'Programming Language :: Python :: 3',
-        'Programming Language :: Python :: 3.4',
-        'Programming Language :: Python :: 3.5',
-        'Programming Language :: Python :: 3.6',
         'Programming Language :: Python :: 3.7',
         'Programming Language :: Python :: 3.8',
         'Programming Language :: Python :: 3.9',
         'Programming Language :: Python :: 3.10',
+        'Programming Language :: Python :: 3.11',
+        'Programming Language :: Python :: 3.12',
         'Topic :: Software Development :: Libraries :: Python Modules',
     ],
     extras_require={
         'dev': [
             'tox',
-            'PyTest             ; python_version >= "3.6"',
-            'PyTest < 5         ; python_version < "3.6"',
-            'PyTest-Cov         ; python_version >= "3.6"',
-            'PyTest-Cov < 2.6   ; python_version < "3.6"',
+            'PyTest',
+            'PyTest-Cov',
             'bump2version < 1',
             'sphinx < 2',
-            # Python 2.7 EOL:
-            'importlib-metadata < 3       ; python_version < "3"',
-            'importlib-resources < 4      ; python_version < "3"',
-            'configparser < 5             ; python_version < "3"',
-            'sphinxcontrib-websupport < 2 ; python_version < "3"',
-            'zipp < 2                     ; python_version < "3"',
+            'importlib-metadata'
```

Comment:
Is this library used anywhere?

Reply:
I didn't actually see it used in the code?

Anyway, one suggestion is to use the stdlib version for 3.8+:

```cfg
  'importlib-metadata; python_version < "3.8"',
```
```python
try:
    # Python 3.8+
    import importlib.metadata as importlib_metadata
except ImportError:
    # Python 3.7
    import importlib_metadata  # type: ignore
```

And then it can be removed when 3.7 is EOL in June: https://devguide.python.org/versions/

Reply:
I failed to document _why_ I added it in 1827126fb1fb8a36073ce00202e4f12afbe5aaaf but I did note that it is needed for dev only. I think maybe wrapt needed it?

```
$ rg importlib
setup.py
187:            'importlib-metadata'
$ rg metadata
setup.py
187:            'importlib-metadata'
```

What's the danger of leaving it now and coming back to removing it? It's a dev dep so it shouldn't affect the public dependencies list.

Reply:
Ya know, I'll bet I had it in there to consolidate the lines removed below, assuming that `importlib-metadata` was actually used somewhere…